src/main/java/net/imagej/ops/image/watershed/WatershedSeeded.java

```diff
@@ -0,0 +1,404 @@
+	private static final AtomicLong seq = new AtomicLong();
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public void compute(final RandomAccessibleInterval<T> in, final ImgLabeling<Integer, IntType> out) {
+		final RandomAccess<T> raIn = in.randomAccess();
+
+		// extend border to be able to do a quick check, if a voxel is inside
+		final ExtendedRandomAccessibleInterval<LabelingType<Integer>, ImgLabeling<Integer, IntType>> outExt = Views
+				.extendBorder(out);
+		final OutOfBounds<LabelingType<Integer>> raOut = outExt.randomAccess();
+
+		// if no mask provided, set the mask to the whole image
+		if (mask == null) {
+			mask = (RandomAccessibleInterval<B>) ops().create().img(in, new BitType());
+			for (B b : Views.flatIterable(mask)) {
+				b.set(true);
+			}
+		}
+
+		// initialize output labels
+		final Cursor<B> maskCursor = Views.flatIterable(mask).cursor();
+		while (maskCursor.hasNext()) {
+			maskCursor.fwd();
+			if (maskCursor.get().get()) {
+				raOut.setPosition(maskCursor);
+				raOut.get().clear();
+				raOut.get().add(INIT);
+			}
+		}
+
+		// RandomAccess for Mask, Seeds and Neighborhoods
+		final RandomAccess<B> raMask = mask.randomAccess();
+		final RandomAccess<LabelingType<Integer>> raSeeds = seeds.randomAccess();
+		final Shape shape;
+		if (useEightConnectivity) {
+			shape = new RectangleShape(1, true);
+		} else {
+			shape = new DiamondShape(1);
+		}
+		final RandomAccessible<Neighborhood<T>> neighborhoods = shape.neighborhoodsRandomAccessible(in);
+		final RandomAccess<Neighborhood<T>> raNeigh = neighborhoods.randomAccess();
+
+		/*
+		 * Carry over the seeding points to the new label and adds them to a
+		 * voxel priority queue
+		 */
+		final PriorityQueue<WatershedVoxel> pq = new PriorityQueue<>();
+
+		// Only iterate seeds that are not excluded by the mask
+		final IterableRegion<B> maskRegions = Regions.iterable(mask);
+		final IterableInterval<LabelingType<Integer>> seedsMasked = Regions.sample(maskRegions, seeds);
+		final Cursor<LabelingType<Integer>> cursorSeeds = seedsMasked.localizingCursor();
+
+		while (cursorSeeds.hasNext()) {
+			final Set<Integer> l = cursorSeeds.next();
+			if (l.isEmpty()) {
+				continue;
+			}
+			if (l.size() > 1) {
+				throw new IllegalArgumentException("Seeds must have exactly one label!");
+			}
+			final Integer label = l.iterator().next();
+			if (label < 0) {
+				throw new IllegalArgumentException("Seeds must have positive integers as labels!");
+			}
+			raNeigh.setPosition(cursorSeeds);
+
+			final Cursor<T> neighborhood = raNeigh.get().cursor();
+
+			// Add unlabeled neighbors to priority queue
+			while (neighborhood.hasNext()) {
+				neighborhood.fwd();
+				raSeeds.setPosition(neighborhood);
+				raMask.setPosition(neighborhood);
+				raOut.setPosition(neighborhood);
+				if (!raOut.isOutOfBounds() && raMask.get().get() && raSeeds.get().isEmpty()
+						&& !raOut.get().contains(INQUEUE)) {
+					raIn.setPosition(neighborhood);
```

Comment:
You can get rid of the line and use `neighborhood.get()` in L212 to get the value from `raIn` at the position of `neighborhood`. In that case, also call `positionToIndex` with `neighborhood` instead of `raIn`.